### ISSUE TITLE
Lack of Documentation for GPU Use, Especially in Metal GPUs in MacBook M1, M1 Max and M2 Chips

### ISSUE THREAD

**Issue by Killpit** [tensorflow/tensorflow#61092]
Hi,

Whenever I tried to use a GPU on MPS with MacBook M1, I generally fail to use the GPU and whenever I tried to reach out to documentation for help, it doesn't provide much help. In addition to the documentation issue, there's a slowdown on M1, M1 Max and M2 chips when I use TensorFlow 2.11+

WARNING:absl:At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.
WARNING:absl:There is a known slowdown when using v2.11+ Keras optimizers on M1/M2 Macs. Falling back to the legacy Keras optimizer, i.e., `tf.keras.optimizers.legacy.Adam`.
Epoch 1/5
2023-06-27 14:57:34.718668: I tensorflow/core/grappler/optimizers/custom_graph_optimizer_registry.cc:114] Plugin optimizer for device_type GPU is enabled.
517/517 [==============================] - ETA: 0s - loss: 0.4559 - accuracy: 0.82332023-06-27 14:58:57.433849: I tensorflow/core/grappler/optimizers/custom_graph_optimizer_registry.cc:114] Plugin optimizer for device_type GPU is enabled.

**Comment by sachinprasadhs** [comment]
Hi,

Thanks for reporting the issue.

Tensorflow Metal support for M1 and M2 were not officially supported by us. 
As shown in the warning, the performance impact for M1 and M2 is shown.

If your issue is specific to the warning you have mentioned, as mentioned in the warning, please use `tf.keras.optimizers.legacy.Adam`, this is the optimizer which was used till 2.10 version until new optimizer was made as default `tf.keras.optimizers.Adam`

**Comment by Killpit** [comment]
Should I use TensorFlow 2.10 or should O write tf.keras.optimizers.legacy.Adam until the problem is solved.

**Comment by sachinprasadhs** [comment]
Yes, you are right, you can use `tf.keras.optimizers.legacy.Adam` until the performance issue related to M1, M2 is solved.
If it is not impacting too much, you can continue using the new `Adam` optimizer

**Comment by Killpit** [comment]
If I'm not mistaken, I should use tf.keras.optimizers.legacy Adam for Tensorflor 2.11+

**Comment by sachinprasadhs** [comment]
Yes, to get the better performance on M1 & M2.

**Comment by Killpit** [comment]
Can we use it for loss functions too?

**Comment by sachinprasadhs** [comment]
Hi, For loss functions, please use one of the https://www.tensorflow.org/api_docs/python/tf/keras/losses
